packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/TypeProvider.cs

```diff
@@ -12,11 +12,7 @@ public abstract class TypeProvider
     {
         protected string? _deprecated;
 
-        private string? _fileName;
-
-        public string FileName => _fileName ??= GetFileName();
-        protected virtual string GetFileName() => $"{Name}.cs";
-
+        public abstract string FileName { get; }
```

Comment:
Can we rename this to RelativeFilePath and add documentation explaining where it is relative from?